src/amo/components/AddonReviewList.js

```diff
@@ -80,40 +85,37 @@ export function loadAddonReviews({ addonSlug, dispatch }) {
       // For example, the user selected a star rating but hasn't submitted
       // review text yet.
       const reviews = allReviews.filter((review) => Boolean(review.body));
-
-      const action = setAddonReviews({ addonSlug, reviews });
-      dispatch(action);
-      return action.payload.reviews;
+      dispatch(setAddonReviews({ addonSlug, reviews }));
     });
 }
 
-export function loadInitialData(
-  { store, params },
-  { _loadAddonReviews = loadAddonReviews } = {},
-) {
-  const { slug } = params;
-  return new Promise((resolve) => {
-    if (!slug) {
-      throw new Error('missing URL param slug (add-on slug)');
-    }
-    resolve();
-  })
-    .then(() => Promise.all([
-      _loadAddonReviews({ addonSlug: slug, dispatch: store.dispatch }),
-      loadAddonIfNeeded({ store, params }),
-    ]))
-    .then(([reviews]) => {
-      const addon = findAddon(store.getState(), slug);
-      const initialData = { addon, reviews };
-      return initialData;
-    });
+export function loadInitialData({ store, params }) {
+  const { addonSlug } = params;
+  if (!addonSlug) {
+    return Promise.reject(new Error('missing URL param addonSlug'));
+  }
+  return Promise.all([
+    loadAddonReviews({ addonSlug, dispatch: store.dispatch }),
+    loadAddonIfNeeded({ store, params: { slug: addonSlug } }),
+  ]);
+}
+
+export function mapStateToProps(state, ownProps) {
+  if (!ownProps || !ownProps.params || !ownProps.params.addonSlug) {
```

Comment:
Yes it could in a situation where the redux state changes after the page initially loads (e.g. someone submits a review which loads it into state).